### PR TITLE
Wait ability end task

### PR DESCRIPTION
**Pull request by Frickerson**

### CODE REVIEW

File: Source/GASExtensions/Private/Tasks/GASExtAT_WaitAbilityEnd.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        OptionalASC = AbilitySystemComponent;
+    }
+
+    DelegateHandle = OptionalASC->OnAbilityEnded.AddUObject( this, &UGASExtAT_WaitAbilityEnd::OnAbilityEnded );
+}
+
+void UGASExtAT_WaitAbilityEnd::OnAbilityEnded( const FAbilityEndedData & ability_ended_data )
+{
+    if ( ability_ended_data.AbilitySpecHandle == AbilitySpecHandle )
```
Comment (TheEmidee) [inline Source/GASExtensions/Private/Tasks/GASExtAT_WaitAbilityEnd.cpp]: look at `UGASExtAT_WaitCharacterLanded`

check `if ( ShouldBroadcastAbilityTaskDelegates() )`

you can also add a flag `bTriggerOnce` amd call `EndTask` if true

Unbind from the event in `OnDestroy`

File: Source/GASExtensions/Public/Tasks/GASExtAT_WaitAbilityEnd.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+{
+    GENERATED_BODY()
+
+public:
+    UFUNCTION( BlueprintCallable, Category = "Ability|Tasks", meta = ( HidePin = "owning_ability", DefaultToSelf = "owning_ability", BlueprintInternalUseOnly = "TRUE" ) )
+    static UGASExtAT_WaitAbilityEnd * WaitAbilityEnd( UGameplayAbility * owning_ability, const FGameplayAbilitySpecHandle & ability_spec_handle, UAbilitySystemComponent * optional_asc = nullptr );
+
+    void Activate() override;
+
+protected:
```
Comment (TheEmidee) [inline Source/GASExtensions/Public/Tasks/GASExtAT_WaitAbilityEnd.h]: can be `private` (especially with a `final` class)

File: Source/GASExtensions/Private/Tasks/GASExtAT_WaitAbilityEnd.cpp (hunk 1 above)
Comment (Frickerson) [reply Source/GASExtensions/Private/Tasks/GASExtAT_WaitAbilityEnd.cpp]: Done.

File: Source/GASExtensions/Public/Tasks/GASExtAT_WaitAbilityEnd.h (hunk 2 above)
Comment (Frickerson) [reply Source/GASExtensions/Public/Tasks/GASExtAT_WaitAbilityEnd.h]: Done.

**Pull request merged**